packages/enzyme-context-react-router-6/src/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState, useEffect } from 'react';
+import { EnzymePlugin, composeWrappingComponents } from 'enzyme-context-utils';
+import { Router } from 'react-router';
+import { createMemoryHistory, Location, MemoryHistory, MemoryHistoryOptions } from 'history';
+
+export type RouterPluginMountOptions = {
+  routerConfig?: MemoryHistoryOptions;
+  location?: Location | string;
+};
+
+export const routerContext: () => EnzymePlugin<RouterPluginMountOptions, MemoryHistory> = () => (
+  node,
+  options,
+) => {
+  const history = createMemoryHistory(options.routerConfig);
+  const RouterContextProvider: React.FC = ({ children }) => {
+    const [currentLocation, setCurrentLocation] = useState(options.location || history.location);
+
+    useEffect(() => {
+      const unlisten = history.listen(({ location }) => {
+        setCurrentLocation(location);
+      });
+      return () => unlisten();
+    }, []);
```

Comment:
Why is this now required?

Reply:
+1 not sure what's going on here. Why aren't we using the built-in `MemoryRouter` component here instead?

Reply:
We could use `MemoryRouter` instead of enzyme-context, which I would prefer for new tests, but this plugin is mainly to facilitate easy migration of existing tests to v6.

"Why is this now required?"
When calling `history.push`, the v6 router does not automatically re-render the location prop - even when it is passed as `history.location`. Because of this, we use a stateful variable and tap into the `history.listen` hook to update that variable in a callback. The `useEffect` is just a mount/unmount for that callback.

Reply:
> We could use MemoryRouter instead of enzyme-context

My question is: why aren't we using `MemoryRouter` in this enzyme-context plugin implementation?

Reply:
What would that look like? My understanding was that enzyme-context is providing a `history` context to the wrapper. The api for `MemoryRouter` doesn't expose this - so the usage of the plugin would change.
https://reactrouter.com/docs/en/v6/routers/memory-router

If we want to migrate existing tests to v6, this implementation means
```
wrapper.history.push('new_path`)
```
will still work.

Reply:
Got it! I didn't realize you couldn't pass your own `history` to `MemoryRouter`.

In that case, just two requests:

* Instead of `useEffect`, can you use `useLayoutEffect` as it better matches what the [real thing does](https://github.com/remix-run/react-router/blob/f9c4a0e8ec022545b2679d381dc41652f1694804/packages/react-router/lib/components.tsx#L49)
* I think we should kill the ability to pass `location`. We should err on the side of having one way to do something, and might as well use the API that already exists!

Reply:
That sounds good. Updated.

Also, it seems that the CI is not being picked up by Travis.
https://app.travis-ci.com/github/trialspark/enzyme-context/branches